bricks/model/__brick__/{{~ copy_with }}

```diff
@@ -1,8 +1,8 @@
 /// Creates a copy of the current {{model_name.pascalCase()}} with property changes
 {{model_name.pascalCase()}} copyWith({{#hasProperties}}{ {{#properties}}
-  {{#hasSpecial}}{{{type}}}{{/hasSpecial}}{{^hasSpecial}}{{type}}{{/hasSpecial}}? {{name}},{{/properties}}
+  {{#hasSpecial}}{{{type}}}{{/hasSpecial}}{{^hasSpecial}}{{type}}{{/hasSpecial}}{{^isNullable}}?{{/isNullable}} {{name}},{{/properties}}
```

Comment:
When we are copyingWith, we should allow developers to pass null to any nullable property in the copyWith method. One solution is to use a function to set the value. This gives you more options.

- A function that isn't provided: `null`
    - This will not change the value
- A function that is provided and returns null: `() => null`
    - This will set the value to `null`
- A function that returns the name: `() => 'Gustavo'`
    - This will set the value to `Gustavo`

So what this may look like now is  `{{#hasSpecial}}{{{type}}}{{/hasSpecial}}{{^hasSpecial}}{{type}}{{/hasSpecial}}{{^isNullable}}?{{/isNullable}}{{isNullable}}Function()?{{/isNullable}} {{name}},{{/properties}}`

This would also update the below `return` to something like `{{name}}: {{name}} != null ? {{name}}() : this.{{name}}`

Reply:
> Everything else looks good besides this 1 change requirement.
> 
> Thanks for this!

Thanks, I've been able to set that up with a response for the model above being
```dart
    /// Creates a copy of the current SampleModel with property changes
  SampleModel copyWith({ 
    String Function()? id,
    String? content,
    DateTime Function()? createdAt,
    DateTime? updatedAt,
    List<String?> Function()? comments,
    List<String?>? sampleNulls,
  }) {
    return SampleModel(
      id: id!=null ? id() : this.id,
      content: content ?? this.content,
      createdAt: createdAt!=null ? createdAt() : this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      comments: comments!=null ? comments() : this.comments,
      sampleNulls: sampleNulls ?? this.sampleNulls,
    );
  }
```

Reply:
Looks great! I have pushed a couple of changes to match the requirements.

Thanks for this! I will merge, and release a new version of the model brick soon!